### ISSUE TITLE
SuperDSC aborts FP8 _scaled_mm at Granite fused N (6144 / 25600) and wide M=16

### ISSUE THREAD

**Issue by GOavi101** [torch-spyre/torch-spyre#4179]
## What

`aten._scaled_mm` with the canonical FP8 graph (`quantize_fp8_with_scale` + `quantize_weight_fp8_with_scale` + `_scaled_mm`) fails to compile on SuperDSC for Granite 8B fused linear shapes. `dynamic=False`. Scales are graph inputs (amax is not fused into `qfp8ch`).

Works: `N ∈ {128, 1024, 4096}` and wide GEMMs at `M ∈ {1, 2, 4}`. Narrow 128-wide at M=128 is fine.

Aborts:

| Case | M | K | N | Notes |
|---|---|---|---|---|
| fused QKV | 1 | 4096 | **6144** | decode; 4096+1024+1024 |
| fused gate_up | 1 | 4096 | **25600** | 2×12800 |
| wide prefill / warmup | **16** | 4096 | 4096 | spyre-inference warmup M |
| bench prefill | 64 | 4096 | 4096 | abort without M-tile of 4 |

`insert_bmm_padding` only pads K. It does not legalize fused N or awkward M.

`qfp8wt` is a 2×64 stick layout. Work division pins the second stick dim and reduction K to `split=1` (`qfp8wt_split_domains` / `qfp8wt_matmul_k_split_domains`). Coarse tiling on main does not make N=6144 or M=16 compile as one `_scaled_mm`.

Eager `float8.to("spyre")` is the wrong layout; the graph must take FP16 `b` and assign `qfp8wt` via `quantize_weight_fp8_with_scale`. The abort is SuperDSC of that fused graph at Granite N/M.

## Why

`ibm-granite/granite-3.3-8b-instruct-FP8` (and 4.x 8B dense) uses those fused widths (hidden=4096, GQA 32/8 × 128, intermediate=12800). spyre-inference currently N-splits to `{4096, 1024, 128}` and M-tiles to 4, then graph-breaks so each tile is its own 3-op compile. That is correct but slow. Once these shapes compile as one `_scaled_mm`, we can drop the tiles.

## Repro

```python
import torch
import torch_spyre  

@torch.compile(backend="inductor", dynamic=False)
def mm(x, sa, w, sb):
    xa = torch.ops.spyre.quantize_fp8_with_scale(x, sa)
    wb = torch.ops.spyre.quantize_weight_fp8_with_scale(w, sb)
    return torch.ops.aten._scaled_mm(
        xa, wb, scale_a=sa, scale_b=sb, out_dtype=torch.float16
    )

def run(m, k, n):
    x = torch.randn(m, k, dtype=torch.float16, device="spyre")
    w = torch.randn(k, n, dtype=torch.float16, device="spyre")
    sa = (x.abs().amax() / 448).to(torch.float16).reshape(1)
    sb = (w.abs().amax() / 448).to(torch.float16).reshape(1)
    print(f"M={m} K={k} N={n}", flush=True)
    y = mm(x, sa, w, sb)
    print("ok", tuple(y.shape))

run(1, 4096, 4096)   # OK
run(1, 4096, 6144)   # abort
# run(16, 4096, 4096)  # abort
# run(1, 4096, 25600)  # abort
```

## Expected

- M=1, K=4096, N=6144 and N=25600 compile and run.
- Wide N=4096 at M=16 (and ideally prefill M=64) compiles without a host M-tile of 4.

**Comment by SudhanvaR** [comment]
Will Start to look into this issue

**Comment by Akash-Nayak** [comment]
I have created a PR to fix this issue- https://github.com/torch-spyre/torch-spyre/pull/4235
